### PR TITLE
tmpfs support

### PR DESCRIPTION
**Pull request by aanand**
Sits on / replaces #939.

- Rebased on master
- Addressed @dnephin's comments about the error messages
- Reworded documentation a little
- Added integration test

### CODE REVIEW

File: tests/integration/container_test.py [hunk 1]
```diff
@@ -382,6 +382,21 @@ def test_create_with_environment_variable_no_value(self):
...
+
+        container = self.client.create_container(
+            BUSYBOX,
+            ['echo'],
+            host_config=self.client.create_host_config(
+                tmpfs=tmpfs))
+
+        self.tmp_containers.append(container['Id'])
+        config = self.client.inspect_container(container)
+        assert config['HostConfig']['Tmpfs'] == tmpfs
```
Comment (dnephin) [inline tests/integration/container_test.py]: This is failing with a KeyError for some reason, I'm not sure why, it seems to be correct to me.

**Comment by dnephin** [comment]
Code looks good, but the test is failing for some reason

**Comment by aanand** [comment]
Ah, it's because of the Engine version (1.9.0). I'll decorate the test.

File: docs/tmpfs.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Using tmpfs
```
Comment (shin-) [inline docs/tmpfs.md]: This file needs to be indexed in `mkdocs.yml`

**Comment by shin-** [comment]
Minor docs comment, apart from that, LGTM.

**Comment by aanand** [comment]
Updated `mkdocs.yml`

**Comment by dnephin** [comment]
LGTM

**Pull request merged**